sklearn/utils/estimator_checks.py

```diff
@@ -468,55 +486,199 @@ def check_transformer_pickle(name, Transformer):
     assert_array_almost_equal(pickled_X_pred, X_pred)
 
 
-def check_estimators_partial_fit_n_features(name, Alg):
-    # check if number of features changes between calls to partial_fit.
-    if not hasattr(Alg, 'partial_fit'):
-        return
+@ignore_warnings
+def check_partial_fit_n_features(name, Estimator):
+    """Check if ValueError is raised if number of features changes between
+    multiple partial_fit calls.
+    """
+    name  # avoids unused argument error
+
     X, y = make_blobs(n_samples=50, random_state=1)
     X -= X.min()
-    with warnings.catch_warnings(record=True):
-        alg = Alg()
-    set_fast_parameters(alg)
-    if isinstance(alg, ClassifierMixin):
-        classes = np.unique(y)
-        alg.partial_fit(X, y, classes=classes)
-    else:
-        alg.partial_fit(X, y)
 
-    assert_raises(ValueError, alg.partial_fit, X[:, :-1], y)
+    estimator = Estimator()
+
+    set_random_state(estimator)
+    set_fast_parameters(estimator)
+
+    classes = np.unique(y)
+    _partial_fit(estimator, X, y, classes)
+    assert_raises(ValueError, _partial_fit, estimator, X[:, :-1], y)
+
+
+@ignore_warnings
+def check_partial_fit_general(name, Estimator):
```

Comment:
I would name this function `check_partial_fit_clone_consistency` to be more specific.